hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsModule.java

```diff
@@ -185,8 +188,9 @@ static ContractValidationRule providePrefixCodeRule() {
     static MessageCallProcessor provideV_0_30MessageCallProcessor(
             final @V_0_30 EVM evm,
             final @V_0_30 PrecompileContractRegistry precompiles,
-            final Map<String, PrecompiledContract> hederaPrecompileList) {
-        return new HederaMessageCallProcessor(evm, precompiles, hederaPrecompileList);
+            final Map<String, PrecompiledContract> hederaPrecompileList,
+            final @Named("HederaSystemAccountDetector") Predicate<Address> hederaSystemAccountDetector) {
```

Comment:
EVM 30 didn't check system accounts, we shouldn't be injecting this to older evm versions.

Reply:
@shemnon are you recommending this be part of the 0.38 new version? and 0.30 and 0.34 preserve the lack of this logic.
Makes sense just want to make sure we're clear.